client/devdocs/design/search-collection.jsx

```diff
@@ -31,49 +34,106 @@ const shouldShowInstance = ( example, filter, component ) => {
 	return ! filter || searchPattern.toLowerCase().indexOf( filter ) > -1;
 };
 
-const Collection = ( { children, filter, section = 'design', component } ) => {
-	let showCounter = 0;
-	const summary = [];
-
-	const examples = React.Children.map( children, example => {
-		if ( ! example || ! shouldShowInstance( example, filter, component ) ) {
-			return null;
-		}
-
-		const exampleName = getComponentName( example );
-		const exampleLink = `/devdocs/${ section }/${ camelCaseToSlug( exampleName ) }`;
-
-		showCounter++;
-
-		if ( filter ) {
-			summary.push(
-				<span key={ `instance-link-${ showCounter }` } className="design__instance-link">
-					<a href={ exampleLink }>{ exampleName }</a>
-					,&nbsp;
-				</span>
+class Collection extends React.PureComponent {
+	static propTypes = {
+		children: PropTypes.node,
+		component: PropTypes.string,
+		examplesToMount: PropTypes.number,
+		filter: PropTypes.string,
+		section: PropTypes.string,
+	};
+
+	static defaultProps = {
+		examplesToMount: 7,
+		section: 'design',
+	};
+
+	_setLazyRenderRef = ref => {
```

Comment:
We usually try to avoid underscore-prefixed names – they don't bring too much meaning.